### DISCUSSION TITLE
Remove luma/chroma mapping in Profile 8.1 from Profile 7 FEL

### DISCUSSION THREAD

**Discussion by aaronmmarks** [quietvoid/dovi_tool#219] [Q&A]
Is it possible to remove luma/chroma mapping for files that were previously converted to Profile 8.1 from Profile 7 FEL with Mode 5 (old Mode 2)? If so, how would I accomplish this? 

I've noticed that Profile 7 FEL files that were converted to Profile 8.1 with Mode 5 (old Mode 2) fallback to HDR in the Infuse 7.5 betas (current 7.5.4370). I don't have access to all the original FELs (deleted to save space), but I have converted a few that I kept around with the new Mode 2 (removing luma/chroma mapping) and these seem to register as DoVi in Infuse 7.5.4370. 

Thank you for the help!

**Comment by aaronmmarks** [comment]
I think I answered my own question in that this can be done by...
1) Demuxing files converted from Profile 7.1 FEL with pre-v2 `dovi_tool` 
2) Create a file named `remove_mapping.json` containing the following text:
```
{
"remove_mapping": true
}
```
3. Run the following command:
```
dovi_tool editor -i .\RPU.bin -j .\remove_mapping.json -o RPU2.bin
```
...which will produce the following:
```
EditConfig {
  "mode": 0,
  "remove_cmv4": false,
  "remove_mapping": true
}
Parsing RPU file...
Removing polynomial/MMR mapping...
Final metadata length: 261792
Writing RPU file...
```
4. Inject the new RPU into the BL.
7. Remux BL+RPU back into the original MKV. 

Any issues with this? 
Is there a faster way to do this?

**Reply by quietvoid** [reply]
You can try
```
dovi_tool --edit-config remove_mapping.json convert video.hevc
```
Then it only needs to be remuxed. Can speed it up by piping the video with FFmpeg.

**Reply by aaronmmarks** [reply]
That worked and sped it up!

Can you think of a way that you could automate the following for a large quantity of files?
1. Demux the HEVC
2. Create the remove_mapping.json file
3. Run the `dovi_tool --edit-config remove_mapping.json convert video.hevc` command against all the demuxed HEVC files. 
4. Remux the new HEVC files with mapping removed back into the new MKVs replacing the original HEVC files. 

Thanks!

**Reply by quietvoid** [reply]
It's entirely possible with scripting and `mkvmerge`/`mkvextract`.

**Reply by aaronmmarks** [reply]
I came up with this so far but I haven't been able to figure out the `mkvmerge` part yet:
```
cmd

for %a ^
in ("*.mkv") ^
do mkvextract "%a" tracks 0:"%a".hevc 

for %a ^
in ("*.hevc") ^
do dovi_tool --edit-config C:\remove_mapping.json convert "%a" -o "%a".mod.hevc
```

**Reply by quietvoid** [reply]
What about
```
mkvmerge -o "output.mkv" "fixed-video.hevc" -D '(' "original-file.mkv" ')'
```
It would replace the video only in the original file.

**Reply by aaronmmarks** [reply]
Thanks @quietvoid! 
I got this working pretty well for now: 
```
for %1 ^
in ("*.mkv") ^
do ( ^
mkvextract "%1" tracks 0:"%~n1".hevc && ^
dovi_tool --edit-config C:\remove_mapping.json convert "%~n1".hevc -o "%~n1".fixed.hevc && ^
mkvmerge -o "%~n1".fixed.mkv -D "%1" --language 0:en --track-name "0:DoVi P8.1" --original-flag 0:yes "%~n1".fixed.hevc --track-order 1:0 )
```
I'm going to try to automate more tomorrow with recursing through directories.
